satellite/audit/reporter.go

```diff
@@ -118,21 +118,21 @@ func (reporter *Reporter) RecordAudits(ctx context.Context, req *Report) (_ *Rep
 // recordAuditFailStatus updates nodeIDs in overlay with isup=true, auditsuccess=false
 func (reporter *Reporter) recordAuditFailStatus(ctx context.Context, failedAuditNodeIDs storj.NodeIDList) (failed storj.NodeIDList, err error) {
 	defer mon.Task()(&ctx)(&err)
-	var errlist errs.Group
-	for _, nodeID := range failedAuditNodeIDs {
-		_, err := reporter.overlay.UpdateStats(ctx, &overlay.UpdateRequest{
+
+	updateRequests := make([]*overlay.UpdateRequest, len(failedAuditNodeIDs))
+	for i, nodeID := range failedAuditNodeIDs {
+		updateRequests[i] = &overlay.UpdateRequest{
 			NodeID:       nodeID,
 			IsUp:         true,
 			AuditSuccess: false,
-		})
-		if err != nil {
-			failed = append(failed, nodeID)
-			errlist.Add(err)
 		}
 	}
-	if len(failed) > 0 {
-		return failed, errs.Combine(Error.New("failed to record some audit fail statuses in overlay"), errlist.Err())
+
+	failed, err = reporter.overlay.BatchUpdateStats(ctx, updateRequests)
+	if err != nil && len(failed) > 0 {
```

Comment:
should this be an `or` not an `and` ?

Reply:
I agree. Updated

Reply:
I think the conditions should be separated. BatchupdateStats returns  nil for error, if the passed slice length is zero, nothing to update and returning nil is ok. but changing the condition to OR will return err.  As suggested above, checking the length of the slice before calling BatchupdateStats will work with using `if err != nil || len(failed) > 0 `